nml/actions/action2layout.py

```diff
@@ -296,6 +296,14 @@ def _validate_sprite(self, name, value):
         else:
             self.sprite_from_action1 = False
             if isinstance(value, expression.ConstantNumeric):
+                # For stations always assume a non-track tile ground, unless explicit rail ground sprite.
+                if (
+                    self.feature == 0x04
+                    and self.type == Action2LayoutSpriteType.GROUND
+                    and value.value != global_constants.constant_numbers["GROUNDSPRITE_RAIL_X"]
+                    and value.value != global_constants.constant_numbers["GROUNDSPRITE_RAIL_Y"]
```

Comment:
To match `SplitGroundSpriteForOverlay` https://github.com/OpenTTD/OpenTTD/blob/7157e9666498d6a500b718a363dbcf16ed62b667/src/station_cmd.cpp#L3025 this should also include the snow variants.
Though NML seems to miss constants for those.